data/server-discovery-and-monitoring/integration/auth-error.yml

```diff
@@ -2,7 +2,10 @@
 runOn:
     # failCommand appName requirements
     - minServerVersion: "4.4"
-      authEnabled: true
+      # The original test has runOn.authEnabled, but the Go driver uses
+      # runOn.auth here in order to use the same RunOnBlock as the other
+      # unified format tests.
+      auth: true
```

Comment:
The original auth-related SDAM spec tests added in [SPEC-1771](https://jira.mongodb.org/browse/SPEC-1771) use `authEnabled` as their run-on requirement for auth. The new unified tests just use `auth` as the key, and because we [use](https://github.com/mongodb/mongo-go-driver/blob/master/mongo/integration/unified_spec_test.go#L49) the mtest's `RunOnBlock` for `mongo/integration/unified_spec_test`, it's probably easier to just change these tests to use the `auth` spelling instead of duplicating the `RunOnBlock` struct and `UnmarshalBSON` method for the older `unified_spec_test`. Let me know what you think.

Reply:
My vote would go for making a change in the Go test runner. The disadvantage of manually modifying these tests is that we would need to make these changes again if these tests change and we need to resync them.

> auth spelling instead of duplicating the RunOnBlock struct

Can the `RunOnBlock` struct have both `auth` and `authEnabled` fields, instead of duplicating the struct?

Reply:
Yep! I've added `authEnabled` to the [`RunOnBlock`](https://github.com/mongodb/mongo-go-driver/blob/master/mongo/integration/mtest/options.go#L53-L60) struct in addition to `auth`. [`verifyAuthConstraint`](https://github.com/mongodb/mongo-go-driver/blob/master/mongo/integration/mtest/mongotest.go#L699-L704) now has to take two parameters: one for `auth` and one for `authEnabled`. When `verifyAuthConstraint` is called [here](https://github.com/mongodb/mongo-go-driver/blob/master/mongo/integration/mtest/mongotest.go#L750) to check if a unit test has specified `Auth(true)` in its setup options, I've just passed `nil` as the second argument. 

This is a _little_ messy, but with the new comment on `verifyAuthConstraint` I think it will probably be more maintainable than altered spec tests.

Reply:
I have a suggestion for changing `verifyAuthConstraint` to accept one bool. Regardless, I agree with it being be more maintainable than altered spec tests.